### ISSUE TITLE
Display issue on snippet edit view

### ISSUE THREAD

**Issue by Quadric** [wagtail/wagtail#5852]
### Issue Summary

When i want to create new instance in snippets panel looks ok. But if i want to edit snippet instance, display is broken because "header" has unnecessary margin bottom 

![image](https://user-images.githubusercontent.com/1465170/75072057-fc4fd600-54f6-11ea-9d15-5827e393fbe3.png)

but on edit panel

![image](https://user-images.githubusercontent.com/1465170/75072103-14bff080-54f7-11ea-8bf9-deeacaaedf4c.png)

### Steps to Reproduce

1. Register model with tabs to snippets
2. Add snippet
3. Edit snippet
4. You'll see that panel is broken

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

I believe the problem is that in file:

```
templates\wagtailsnippets\snippets\create.html
```

there is invocation:

```
    {% include "wagtailadmin/shared/header.html" with title=new_str subtitle=model_opts.verbose_name icon="snippet" tabbed=1 merged=1 %}
```

look for param "merged=1"

but in file:

```
emplates\wagtailsnippets\snippets\edit.html
```

there is invocation

```
    {% include "wagtailadmin/shared/header.html" with title=editing_str subtitle=instance icon="snippet" tabbed=1 %}
```

so it's missing a "merged=1" parameter which will not set class "merged" with "margin-bottom" set to "0" to disable unnecessary margin

### Technical details

* Python version: 3.6.4.
* Django version: 2.2.10
* Wagtail version: 2.8
* Browser version: Google Chrome  79.0.3945.130

**Comment by loicteixeira** [comment]
Thanks for the report @Quadric. By the looks of it, you pinpointed where the problem lies. Would you want to try to submit a PR with the fix you describe?

**Comment by gasman** [comment]
Fixed in #5883
